### PR TITLE
Fix: Text color of the Button not applied, when text-transform has be…

### LINKED ISSUE #2601: Text color of the Button not applied, when `text-transform` has been set.
In case you want to change text `color` of a `Button`, while `text-transform` has been set to uppercase or lowercase, the color of the text would not be applied. This issue is reproducible only for iOS. I am attaching sample code:

main-page.xml

```
<Page xmlns="http://schemas.nativescript.org/tns.xsd">
  <StackLayout>
    <Label text="labelLabel" style="text-transform: uppercase; color:red;"/>
    <Button text="buttonButton" style=" color:red; text-transform: uppercase; " />
    <TextField text="textField" style="text-transform: uppercase; color:red;"/>
    <TextView text="textView" style="text-transform: uppercase;  color:red;" />
  </StackLayout>
</Page>
```

tns-core-modules:2.2.0
tns-ios:2.2.1
CLI:  2.2.1

### PR DESCRIPTION
**Pull request by hamorphis**
…en set.

Resolves #2601

### CODE REVIEW

**Comment by hamorphis** [comment]
uitests

**Comment by hamorphis** [comment]
uitests

File: tns-core-modules/ui/button/button.ios.ts [hunk 1]
```diff
@@ -295,22 +294,28 @@ export class ButtonStyler implements style.Styler {
                 if (hasLetterSpacing) {
                     dict.set(NSKernAttributeName, letterSpacing * button.ios.font.pointSize);
                 }
+            }
 
+            if (button.style.color && button.style.color.ios){
```
Comment (hshristov) [inline tns-core-modules/ui/button/button.ios.ts]: Cache button.style.color

**Comment by hamorphis** [comment]
uitests

**Pull request merged**